lib/synapse/config_generator/haproxy.rb

```diff
@@ -1005,7 +1028,8 @@ def generate_backend_stanza(watcher, config)
         log.debug "synapse: no backends found for watcher #{watcher.name}"
       end
 
-      keys = case watcher.haproxy['backend_order']
+      watcher_haproxy = watcher.generator_config['haproxy']
```

Comment:
I ended up going with watcher_config, but feel free to rename to opts if you want (opts and config are hella confusing no matter how we slice it :-/)